src/etc/inc/util.inc

```diff
@@ -2553,7 +2553,35 @@ function validateipaddr(&$addr, $type, $label, &$err_msg, $alias=false) {
 	return false;
 }
 
-/* returns true if $dhcp6duid is a valid duid entrry */
+/* format a string to look (more) like the expected DUID format:
+ * 1) Replace any "-" with ":"
+ * 2) If the user inputs 14 components, then add the expected "0e:00:" to the front.
+ *    This is convenience, because the actual DUID (which is reported in logs) is the last 14 components.
+ * 3) If any components are input with just a single char (hex digit hopefully), put a "0" in front.
+ *
+ * The final result should be closer to:
+ *
+ * "0e:00:00:01:00:01:nn:nn:nn:nn:nn:nn:nn:nn:nn:nn"
+ *
+ * This function does not validate the input. is_duid() will do validation.
+*/
+function format_duid($dhcp6duid) {
+	$formatted_dhcp6duid = strtolower(str_replace("-", ":", $dhcp6duid));
+	$values = explode(":", $formatted_dhcp6duid);
+	if (count($values) == 14) {
+		$formatted_dhcp6duid = '0e:00:' . $formatted_dhcp6duid;
+		$values = explode(":", $formatted_dhcp6duid);
+	}
+	foreach ($values as $idx => $value) {
+		if (strlen($value) == 1) {
```

Comment:
This length test/add 0 block could be replaced by a single call to `str_pad()` or a `sprintf()` call with a format that would always output two digits.